app/controllers/admin/api/services/proxy/policies_controller.rb

```diff
@@ -40,9 +39,7 @@ def show
   ##~ op.parameters.add name: "policies_config", description: "Proxy policies chain", dataType: "string", paramType: "query", required: true
   #
   def update
-    if proxy.update_attributes(proxy_params)
-      proxy.deploy_v2 if proxy.apicast_configuration_driven
-    end
+    proxy.deploy if proxy.update_attributes(proxy_params)
```

Comment:
I see behaviour changing a bit here. What happens now if proxy is not configuration driven?

Reply:
I know policies don't exist in this case, but does the endpoint prevent it?

Reply:
Although it looks like a change, it's been always evaluated inside `#deploy`. Here, this line: https://github.com/3scale/porta/pull/1470/files#diff-7cad44f925331c18a0c6ddeaeb8126e6R25-R31
I might have overlooked something but I'm positive it's the same behavior, only a couple of duplicated lines less.

Reply:
Consider

```ruby
proxy.update_attributes(proxy_params) == true
proxy.apicast_configuration_driven == false
proxy.service_mesh_integration? == false
```

What happened before and what happens now? Was `proxy.deploy_v1` invoked before? Will it be invoked now?

Reply:
OK gotcha, I mixed it up with the other controller

Reply:
How could I include this in the tests?

Reply:
In fact if we could have `Proxy#deployable?` and use it in the service.
This way you remove complexity and coupling.

You just use the service instead of `proxy.deploy` and can test it inside the service only

```ruby
class ProxyDeploymentService
   def initialize(proxy)
     @proxy = proxy
     # etc ...
   end
end

```

Reply:
OK that's one missing test but the problem with this line of code is that it calls `deploy_v2` only, not `deploy_v1` or `deploy_production_v2`. And `service.call` runs at least one of them, which is way we need another method.